tests/build_/test_arrangements.py

```diff
@@ -496,3 +486,232 @@ def test_koji_parent_in_orchestrator(self, osbs, base_image, scratch,
+            ],
+
+            'prepublish_plugins': [
+            ],
+
+            'exit_plugins': [
+                'koji_import',
+                'koji_tag_build',
+                'store_metadata_in_osv3',
+                'sendmail',
+                'remove_built_image',
+            ],
+        },
+
+        # Changing this? Add test methods
+        WORKER_INNER_TEMPLATE: {
+            'prebuild_plugins': [
+                'add_filesystem',
+                'pull_base_image',
+                'add_labels_in_dockerfile',
+                'change_from_in_dockerfile',
+                'add_help',
+                'add_dockerfile',
+                'distgit_fetch_artefacts',
+                'fetch_maven_artifacts',
+                'koji',
+                'add_yum_repo_by_url',
+                'inject_yum_repo',
+                'distribution_scope',
+            ],
+
+            'buildstep_plugins': [
+            ],
+
+            'postbuild_plugins': [
+                'all_rpm_packages',
+                'tag_by_labels',
+                'tag_from_config',
+                'tag_and_push',
+                'pulp_push',
+                'pulp_sync',
+                'compress',
+                'koji_upload',
+                'pulp_pull',
+            ],
+
+            'prepublish_plugins': [
+                'squash',
+            ],
+
+            'exit_plugins': [
+                'delete_from_registry',
+                'store_metadata_in_osv3',
+            ],
+        },
+    }
+
+    def test_is_default(self):
+        """
+        Test this is the default arrangement
+        """
+
+        # Note! If this test fails it probably means you need to
+        # derive a new TestArrangementV[n] class from this class and
+        # move the method to the new class.
+        assert DEFAULT_ARRANGEMENT_VERSION == self.ARRANGEMENT_VERSION
+
+    @pytest.mark.parametrize('scratch', [False, True])  # noqa:F811
+    def test_koji_upload(self, osbs, scratch):
+        additional_params = {
+            'git_uri': TEST_GIT_URI,
+            'git_ref': TEST_GIT_REF,
+            'user': "john-foo",
+            'component': TEST_COMPONENT,
+            'registry_uris': [],
+            'openshift_uri': "http://openshift/",
+            'builder_openshift_url': "http://openshift/",
+            'build_image': None,
+            'base_image': 'fedora:latest',
+            'name_label': 'fedora/resultingimage',
+            'registry_api_versions': ['v1'],
+            'kojihub': 'http://hub/',
+            'koji_upload_dir': 'upload',
+        }
+        if scratch:
+            additional_params['scratch'] = True
+        params, build_json = self.get_worker_build_request(osbs, additional_params)
+        plugins = get_plugins_from_build_json(build_json)
+
+        with pytest.raises(NoSuchPluginException):
+            get_plugin(plugins, 'exit_plugins', 'koji_promote')
+
+        if scratch:
+            try:
```

Comment:
tab the `return` statement to the left. Because `get_plugin` raises an exception, it never actually hits the `return` statement, which causes the test to continue, triggering the issue you're seeing.